src/main/java/net/dv8tion/jda/internal/interactions/command/localization/LocalizationMapper.java

```diff
@@ -107,10 +110,25 @@ private void forObjects(DataArray source, Function<DataObject, String> keyExtrac
             for (int i = 0; i < source.length(); i++)
             {
                 final DataObject item = source.getObject(i);
-                final String key = keyExtractor.apply(item);
-                keyComponents.push(key);
-                consumer.accept(item);
-                keyComponents.pop();
+                final Runnable runnable = () ->
+                {
+                    final String key = keyExtractor.apply(item);
+                    keyComponents.push(key);
+                    consumer.accept(item);
+                    keyComponents.pop();
+                };
+
+                //We need to differentiate subcommands/groups from options before inserting the "options" separator
+                if (OptionType.fromKey(item.getInt("type", -1)).isOption()) { //-1 when the object isn't an option
```

Comment:
I don't really like this.
I think we should drop the `isOption` method and instead explicitly check here if it is a SUB_COMMAND or SUB_COMMAND_GROUP instead.

```suggestion
                OptionType type = OptionType.fromKey(item.getInt("type", -1));
                bool isOption = type != OptionType.SUB_GROUP_COMMAND && type != OptionType.SUB_COMMAND;
                if (isOption) {
```